components/kubeflow/katib-launcher/src/launch_study_job.py

```diff
@@ -0,0 +1,161 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import argparse
+import datetime
+import json
+import os
+import logging
+import requests
+import subprocess
+import yaml
+import grpc
+
+import api_pb2
+import api_pb2_grpc
+
+from kubernetes import client as k8s_client
+from kubernetes import config
+import study_job_client
+
+def yamlOrJsonStr(str):
+    if str == "" or str == None:
+        return None
+    try:
+        return json.loads(str)
+    except:
+        return yaml.load(str)
+
+def strToList(str):
+    return str.split(",")
+
+def _update_or_pop(spec, name, value):
+    if value:
+        spec[name] = value
+    else:
+        spec.pop(name)
+
+def _generate_studyjob_yaml(src_filename, name, namespace, optimizationtype, objectivevaluename, optimizationgoal, requestcount,
+                            metricsnames, parameterconfigs, nasConfig, workertemplatepath, mcollectortemplatepath, suggestionspec):
+  """_generate_studyjob_yaml generates studyjob yaml file based on hp.template.yaml"""
+  with open(src_filename, 'r') as f:
+    content = yaml.load(f)
```

Comment:
Maybe you can just fully build the spec dict here in code instead of loading it from file and replacing some values?

Reply:
just follow how tfjob-launcher fills spec dict. I will submit another PR to fix the both two components. Thanks!